src/core/server/server.api.md

```diff
@@ -381,6 +382,239 @@ export class Router {
     routes: Array<Readonly<RouterRoute>>;
     }
 
+// @public (undocumented)
+export interface SavedObject<T extends SavedObjectAttributes = any> {
+    // (undocumented)
+    attributes: T;
+    // (undocumented)
+    error?: {
+        message: string;
+        statusCode: number;
+    };
+    // (undocumented)
+    id: string;
+    // (undocumented)
+    migrationVersion?: SavedObjectsMigrationVersion;
+    // (undocumented)
+    references: SavedObjectReference[];
+    // (undocumented)
+    type: string;
+    // (undocumented)
+    updated_at?: string;
+    // (undocumented)
+    version?: string;
+}
+
+// @public (undocumented)
+export interface SavedObjectAttributes {
+    // (undocumented)
+    [key: string]: SavedObjectAttributes | string | number | boolean | null;
+}
+
+// @public
+export interface SavedObjectReference {
+    // (undocumented)
+    id: string;
+    // (undocumented)
+    name: string;
+    // (undocumented)
+    type: string;
+}
+
+// @public (undocumented)
+export interface SavedObjectsBaseOptions {
+    namespace?: string;
+}
+
+// @public (undocumented)
+export interface SavedObjectsBulkCreateObject<T extends SavedObjectAttributes = any> {
+    // (undocumented)
+    attributes: T;
+    // (undocumented)
+    id?: string;
+    // (undocumented)
+    migrationVersion?: SavedObjectsMigrationVersion;
+    // (undocumented)
+    references?: SavedObjectReference[];
+    // (undocumented)
+    type: string;
+}
+
+// @public (undocumented)
+export interface SavedObjectsBulkGetObject {
+    fields?: string[];
+    // (undocumented)
+    id: string;
+    // (undocumented)
+    type: string;
+}
+
+// @public (undocumented)
+export interface SavedObjectsBulkResponse<T extends SavedObjectAttributes = any> {
+    // (undocumented)
+    saved_objects: Array<SavedObject<T>>;
+}
+
+// @public (undocumented)
+export interface SavedObjectsBulkResponse<T extends SavedObjectAttributes = any> {
+    // (undocumented)
+    saved_objects: Array<SavedObject<T>>;
+}
+
+// @internal (undocumented)
+export class SavedObjectsClient {
+    // Warning: (ae-forgotten-export) The symbol "SavedObjectsRepository" needs to be exported by the entry point index.d.ts
+    constructor(repository: SavedObjectsRepository);
+    bulkCreate<T extends SavedObjectAttributes = any>(objects: Array<SavedObjectsBulkCreateObject<T>>, options?: SavedObjectsCreateOptions): Promise<SavedObjectsBulkResponse<T>>;
+    bulkGet<T extends SavedObjectAttributes = any>(objects?: SavedObjectsBulkGetObject[], options?: SavedObjectsBaseOptions): Promise<SavedObjectsBulkResponse<T>>;
+    create<T extends SavedObjectAttributes = any>(type: string, attributes: T, options?: SavedObjectsCreateOptions): Promise<SavedObject<T>>;
+    delete(type: string, id: string, options?: SavedObjectsBaseOptions): Promise<{}>;
+    // (undocumented)
+    errors: {
+        isSavedObjectsClientError: typeof import("./lib/errors").isSavedObjectsClientError;

```

Comment:
1. I've addressed this by creating a `SavedObjectsErrorHelpers` class so that the type documentation is better.
2. Because of the way I type `SavedObjectsClientContract` the static `errors` property isn't included in the type. So when I remove the `public errors` there's a lot of type errors even though this should technically resolve to the static function. I can probably fix this by creating a `SavedObjectsClientContract` interface and duplicating the types from the class. However, because `SavedObjectsErrorHelpers` is just static functions, I've been thinking of removing these convenience `errors` properties and instead let consumers rather import `SavedObjectsErrorHelpers` directly. Because this is a breaking API change, I'd prefer to do this in a separate PR.